12-watchtower-API.md

```diff
@@ -22,227 +22,311 @@ The scope of this bolt does not include:
  - WatchTower server discovery.
  
 
-## Table of Contents
+## Table of Contents 
+* [WatchTower discovery](#watchtower-discovery)
+* [WatchTower services](#watchtower-discovery)
+	* [Basic Service](#basic-service)
+	* [Quality of Service](#quality-of-service)
+* [Sending and receiving appointments](#sending-and-receiving-appointments)
+ 	* [The `appointment` message](#the-appointment-message)
+ 	* [The `appointment_accepted` message](#the-appointment_accepted-message)
+ 	* [The `appointment_rejected` message](#the-appointment_rejected-message)
+* [Quality of Service data](#quality-of-service-data)
+	* [`accountability`](#accountability)
+* [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
+* [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
+* [Payment Modes](#payment-modes)
+* [No compression of justice transaction](#no-compression-of-justice-transaction)
 
-  * [Transaction Locator and Encryption Key](#transaction-locator-and-encryption-key)
-  * [Encryption Algorithms and Parameters](#encryption-algorithms-and-parameters)
-  * [WatchTower API](#WatchTower-api)
-  * [Signed Receipt Fields](#signed-receipt-fields)
-  * [Payment Modes](#payment-modes)
-  * [Number of updates](#number-of-updates)
-  * [No compression of justice transaction](#no-compression-of-justice-transaction)
+## WatchTower discovery
+At this point we're leaving the client/server connection to be protocol agnostic. How the Lightning node finds the WatchTower or how the WatchTower announces their presence and services provided is not specified.
 
-## Connection establishment
-Connections between the client and the server can be long-lived or restarted for every single appointment.
+Therefore, we are assuming that the client and server are connected and that the client have learnt what Quality of Service (`qos`) the tower is offering.
 
-		+-------+                      +-------+
-		|   A   |--(1)--- wt_init ---->|   B   |
-		|       |<-(2)--- wt_init -----|       |
-		+-------+                      +-------+
+## WatchTower services
+
+### Basic Service
+The basic service provided by a WatchTower refers to providing justice transaction relay upon seeing a breach on the blockchain with no finality guarantees (i.e. the transaction inclusion is not guaranteed).
+
+### Quality of Service
```

Comment:
The quality of service ('qos') extends the basic service as the WatchTower provides a signed receipt during the acknowledgement. It is an optional add-on. 

We define it as a single type of 'qos' : 'accountability'

Reply:
I guess main point to say here

"The main point is to add publicly verifiable inside the protocol, that the tower has accepted the job". that is the main point behind it.